WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -12,8 +12,6 @@ import WordPressFlux
 ///   - This controller uses MULTIPLE NSManagedObjectContexts to manage syncing and state.
 ///     - The topic exists in the main context
 ///     - Syncing is performed on a derived (background) context.
-///     - Content is fetched on a child context of the main context.  This allows
```

Comment:
I've removed the `newMainContextChildContext` from this class, replacing it with the stack's main context for now.

I've used this on my iPhone 6s+ and not noted any performance hiccups from it – I suspect that's because this approach may not work anymore? If I'm following this correctly, it looks like this class was listening for when the `mainContext` is saved, and when that happened, it would update the child context. But then it's unclear to me what would cause the `mainContext` to be saved. AFAICT, that would most likely happen in the `ContextManager` as a result of an `NSManagedObjectContextDidSave` notification. But if that's the case, wouldn't changes in the sync context propagate directly to the child context, just with the middle step of the `mainContext` in the middle?

If that's the case, it'd explain why there was no performance difference removing the child context, but I'm not entirely certain I'm understanding this correctly.

Reply:
I'm guessing a lot has changed since that mechanism was first introduced [six years ago](https://github.com/wordpress-mobile/WordPress-iOS/pull/4225). I'm struggling to remember the exact problem it was supposed to solve.  As long as we're not seeing any obvious visual glitch related to syncing and reloading the list (while scrolling) it's probably fine. :)